cmake/tcmalloc.h.in

```diff
@@ -137,7 +141,7 @@ extern "C" {
   PERFTOOLS_DLL_DECL void tc_deletearray_nothrow(void* p,
                                                  const std::nothrow_t&) PERFTOOLS_NOTHROW;
 
-#if @HAVE_STD_ALIGN_VAL_T@ && __cplusplus >= 201703L
+#ifdef PERFTOOLS_HAVE_ALIGNED_NEW
```

Comment:
Note we will be losing a useful check here.

Currently the generated header file will match what is shipped in the tcmalloc binary.
With this change said check will be deferred to link time or even potentially startup time.

A solution would be to unconditionally export new variants of those functions by replacing `align_val_t` with our own type.
Then redefine the existing `align_val_t` functions based on those.

Reply:
Maybe I misunderstood, but I don't agree with playing games trying to mimic/re-implement align_val_t type.

But I agree with your first thesis that this commit is somewhat risky move.

Reply:
I was basically suggesting
```
namespace tcmalloc { enum class align_val_t : std::size_t {}; }

PERFTOOLS_DLL_DECL void* tc_new_aligned2(size_t size, tcmalloc::align_val_t al);
...

#ifdef PERFTOOLS_HAVE_ALIGNED_NEW
  inline void* tc_new_aligned(size_t size, std::align_val_t al) {
      tc_new_aligned2(size,  tcmalloc::align_val_t((std::size_t)al));
  }
  ...
#endif
```

The idea is we make sure the aligned new/delete functions will always be available regardless of the producer's toolchain.

Downsides are - (1) We'll be repeating ourselves, (2) ABI break